exec/java-exec/src/main/java/org/apache/drill/exec/expr/stat/ParquetIsPredicate.java

```diff
@@ -62,90 +60,89 @@ private ParquetIsPredicate(LogicalExpression expr, BiPredicate<Statistics<C>, Ra
     );
   }
 
   /**
    * IS TRUE predicate.
    */
-  private static LogicalExpression createIsTruePredicate(LogicalExpression expr) {
-    return new ParquetIsPredicate<Boolean>(expr,
-        //if max value is not true or if there are all nulls  -> canDrop
-        (exprStat, evaluator) -> !((BooleanStatistics)exprStat).getMax() || isAllNulls(exprStat, evaluator.getRowCount())
-    );
+  private static <C extends Comparable<C>> LogicalExpression createIsTruePredicate(LogicalExpression expr) {
+    return new ParquetIsPredicate<C>(expr,
+      (exprStat, evaluator) -> {
+        if (isAllNulls(exprStat, evaluator.getRowCount()) || (exprStat.genericGetMin().equals(Boolean.FALSE) && exprStat.genericGetMax().equals(Boolean.FALSE))) {
+          return RowsMatch.NONE;
+        }
+        return exprStat.genericGetMin().equals(Boolean.TRUE) && exprStat.genericGetMax().equals(Boolean.TRUE) ? checkNull(exprStat) : RowsMatch.SOME;
+      });
   }
 
   /**
    * IS FALSE predicate.
    */
-  private static LogicalExpression createIsFalsePredicate(LogicalExpression expr) {
-    return new ParquetIsPredicate<Boolean>(expr,
-        //if min value is not false or if there are all nulls  -> canDrop
-        (exprStat, evaluator) -> ((BooleanStatistics)exprStat).getMin() || isAllNulls(exprStat, evaluator.getRowCount())
+  private static <C extends Comparable<C>> LogicalExpression createIsFalsePredicate(LogicalExpression expr) {
+    return new ParquetIsPredicate<C>(expr,
+      (exprStat, evaluator) -> !exprStat.genericGetMin().equals(Boolean.FALSE) || isAllNulls(exprStat, evaluator.getRowCount()) ? RowsMatch.NONE : checkNull(exprStat)
     );
   }
 
   /**
    * IS NOT TRUE predicate.
    */
-  private static LogicalExpression createIsNotTruePredicate(LogicalExpression expr) {
-    return new ParquetIsPredicate<Boolean>(expr,
-        //if min value is not false or if there are no nulls  -> canDrop
-        (exprStat, evaluator) -> ((BooleanStatistics)exprStat).getMin() && hasNoNulls(exprStat)
+  private static <C extends Comparable<C>> LogicalExpression createIsNotTruePredicate(LogicalExpression expr) {
+    return new ParquetIsPredicate<C>(expr,
+      (exprStat, evaluator) -> !exprStat.genericGetMin().equals(Boolean.FALSE) && hasNoNulls(exprStat) ? RowsMatch.NONE : checkNull(exprStat)
     );
   }
 
   /**
    * IS NOT FALSE predicate.
    */
-  private static LogicalExpression createIsNotFalsePredicate(LogicalExpression expr) {
-    return new ParquetIsPredicate<Boolean>(expr,
-        //if max value is not true or if there are no nulls  -> canDrop
-        (exprStat, evaluator) -> !((BooleanStatistics)exprStat).getMax() && hasNoNulls(exprStat)
+  private static <C extends Comparable<C>> LogicalExpression createIsNotFalsePredicate(LogicalExpression expr) {
+    return new ParquetIsPredicate<C>(expr,
+      (exprStat, evaluator) -> !exprStat.genericGetMax().equals(Boolean.TRUE) && hasNoNulls(exprStat) ? RowsMatch.NONE : checkNull(exprStat)
```

Comment:
Why is it necessary to compare `genericGetMax()` to `Boolean.TRUE` when `getMax()` already returns `boolean`?

Reply:
Because here we use Statistics class, not BooleanStatistics. GetMax belongs to the later. It avoids an explicit cast.

Reply:
"is true" can only be applied to a boolean expression and boolean expression can only produce `BooleanStatistics`. It is invalid anyway to compare `Integer`, `Long` or `Double` to `True` or `False`.

Reply:
Done